acceptance/tests/resource/mount/query.rb

```diff
@@ -1,35 +1,32 @@
 test_name "should be able to find an existing filesystem table entry"
 
 confine :except, :platform => ['windows']
-confine :except, :platform => /solaris/
-confine :except, :platform => /aix/
+confine :except, :platform => /osx/ # See PUP-4823
+confine :except, :platform => /solaris/ # See PUP-5201
+
+require 'puppet/acceptance/mount_utils'
+extend Puppet::Acceptance::MountUtils
 
-fstab = '/etc/fstab'
 name = "pl#{rand(999999).to_i}"
 
 agents.each do |agent|
+  fs_file = filesystem_file(agent)
+
   teardown do
     #(teardown) restore the fstab file
-    on(agent, "mv /tmp/fstab #{fstab}", :acceptable_exit_codes => (0..254))
-    #(teardown) umount disk image
-    on(agent, "umount /#{name}", :acceptable_exit_codes => (0..254))
-    #(teardown) delete disk image
-    on(agent, "rm /tmp/#{name}", :acceptable_exit_codes => (0..254))
-    #(teardown) delete mount point
-    on(agent, "rm -fr /#{name}", :acceptable_exit_codes => (0..254))
+    on(agent, "mv /tmp/fs_file_backup #{fs_file}", :acceptable_exit_codes => (0..254))
   end
 
   #------- SETUP -------#
-  step "(setup) backup fstab file"
-  on(agent, "cp #{fstab} /tmp/fstab", :acceptable_exit_codes => [0,1])
+  step "(setup) backup #{fs_file} file"
+  on(agent, "cp #{fs_file} /tmp/fs_file_backup", :acceptable_exit_codes => [0,1])
 
   step "(setup) add entry to filesystem table"
-  on(agent, "echo '/tmp/#{name}  /#{name}  0  0' >> #{fstab}")
+  add_entry_to_filesystem_table(agent, name)
 
   #------- TESTS -------#
   step "verify mount with puppet"
   on(agent, puppet_resource('mount', "/#{name}")) do |res|
-    fail_test "didn't find the mount #{name}" unless res.stdout.include? "#{name}"
+    fail_test "didn't find the mount #{name}" unless res.stdout.match(/'\/#{name}':\s+ensure\s+=>\s+'unmounted'/)
```

Comment:
@johnduarte that was actually one of the original issues with this test - we never mount the filesystem (or even create it). We're just adding it the the filesystem table and seeing if puppet knows about it. Since it's not mounted, puppet will return with `unmounted`. It's necessary to check for that, though, because you can ask puppet about any non-existent mount resource and it will return with,

```
mount { 'nonexist':
    ensure => absent
}
```
 - meaning that this test would always pass, even if the entry wasn't in the fstab.